.github/workflows/formatting.yml

```
@@ -10,8 +10,32 @@ jobs:
   formatting:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2 # v2 minimum required
-      - uses: axel-op/googlejavaformat-action@v3
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
         with:
-          commitMessage: "chore: Google Java Format"
-          githubToken: ${{ secrets.GITHUB_TOKEN }}
+          java-version: 11
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Check code compliance to google java format standards
+        uses: eskatos/gradle-command-action@v1
+        env:
+          versionTag: ${{ steps.prep.outputs.versionTag }}
+        with:
+          arguments: verGJF
+      - name: Comment PR if code does not comply to Google Java style guide
+        if: ${{ failure() }}
+        uses: actions/github-script@v4
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            github.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: "This contribution does not follow the conventions set by the [Google Java style guide](https://google.github.io/styleguide/javaguide.html). Please [download the formatter](https://github.com/google/google-java-format/releases) and run it with ```java -jar /path/to/google-java-format-1.11.0-all-deps.jar -i $(find . -name '*.java')``` to fix formatting errors."
```

Actually, if we're including a dependency on `com.github.sherter.google-java-format` as part of the project, can't the developer just run `./gradlew goJF` themselves to format the files without needing to download the Google JAR separately?

An alternate approach here is to always run this Gradle task ourselves for every Gradle build, to ensure that any contributions are formatted before the developer commits the files locally. This is definitely more convenient, but I don't know what the build performance hit would be. I believe you could filter the input so that only changed files are formatted on every build, to reduce the processing on each build.

---

> Actually, if we're including a dependency on com.github.sherter.google-java-format as part of the project, can't the developer just run ./gradlew goJF themselves to format the files without needing to download the Google JAR separately?

Indeed!

An alternate approach here is to always run this Gradle task ourselves for every Gradle build, to ensure that any contributions are formatted before the developer commits the files locally. This is definitely more convenient, but I don't know what the build performance hit would be. I believe you could filter the input so that only changed files are formatted on every build, to reduce the processing on each build, if that's an issue.
```suggestion
              body: "This contribution does not follow the conventions set by the [Google Java style guide](https://google.github.io/styleguide/javaguide.html). Please [download the formatter](https://github.com/google/google-java-format/releases) and run it with ```java -jar /path/to/google-java-format-1.11.0-all-deps.jar -i $(find . -name '*.java')``` to fix formatting errors."
```
Are you talking a sort of pre-commit pre hook?

---

No, you can tell Gradle to run a task as the final stage in the build pipeline, and the Google Java format task would be the final task.

See https://barbeau.medium.com/how-to-avoid-duplicate-code-for-similar-sdks-7e7812e67708 for an example. In this example, I needed to copy source code files to a different package and use a regex to replace some class names and packages on each build. This way the same source code could be used to support different SDKs that had *slight* differences in the API implements (e.g., Google Maps vs Amazon Maps). So when you tell Gradle to build, it executes the normal build and then these tasks as the end.

---

Got it! As discussed we will implement the gradle task you mentioned in a separate PR so that we can handle the case where someone would submit code via the GitHub web editor interface.